pkgs/native_assets_builder/test_data/treeshaking_native_libs/hook/build.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:logging/logging.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+void main(List<String> arguments) async {
+  await build(arguments, (config, output) async {
+    final cbuilder = CBuilder.library(
+      name: config.packageName + (config.linkingEnabled ? '_static' : ''),
+      assetName: config.linkingEnabled
+          ? 'somename'
```

Comment:
Why not have the correct name and reuse the correct name in the link hook:

```dart
assetName: config.assets.single.id.split('/').skip(1).join('/'),
```

Maybe we should expose the `name` and `package` of `Asset`? Not just the `id`?


The link hook should be written in a way that it makes a dylib out of every individual static library. (Otherwise the asset id's will be one and the same if coming from the link hook and different if coming from the build hook if linking is not enabled.)